File: homework/hw2/code/documentary.cpp
#include "documentary.h"

documentary::documentary(char *name, int name_size,
                         int release_year, int duration) : movie(name, name_size, release_year) {
    this->duration = duration;
}

documentary::~documentary() {
    delete[] name;
}

void documentary::print(FILE *out) {
    fprintf(out, "[Documentary]\n"
                 "-name: %s\n"
                 "-release year: %d\n"
                 "-duration: %d\n"
                 "-function result: %f\n\n", name, release_year, duration, year_div_name());
}

File: homework/hw3/code/documentary.py
from movie import Movie
import random


class Documentary(Movie):
    def __init__(self, *args):
        if len(args) > 0:
            super(Documentary, self).__init__(*args[:-1])
            self._duration = args[-1]
        else:
            super(Documentary, self).__init__()
            self._duration = random.randint(1950, 2022)

    def __str__(self):
        return """[Documentary]:
              -name: {0}
              -release year: {1}
              -duration: {2}
              -function result: {3}""".format(self._name, self._release_year,
                                              self._duration, self._release_year_div_name())
